pkg/infra/tower.go

```diff
@@ -98,11 +102,59 @@ func (t *Tower) reshuffle(numOfFloors int) {
 }
 
 func (t *Tower) hpDecay() {
-	// TODO: can add a parameter
 	for _, agent := range t.agents {
-		newHP := agent.hp - 20
+		newHP := 0
+		switch {
+		case agent.hp >= t.healthInfo.StrongLevel:
+			if agent.hp >= t.healthInfo.StrongLevel+int(0.63*float64(t.healthInfo.WidthStrong)) {
+				newHP = int(math.Max(float64(t.healthInfo.StrongLevel), float64(agent.hp)-0.63*float64(t.healthInfo.WidthStrong)))
+				agent.daysInState++
+			} else {
+				newHP = int(math.Max(float64(t.healthInfo.HealthyLevel), float64(agent.hp)-0.63*float64(t.healthInfo.WidthStrong)))
+				agent.daysInState = 0
+			}
+
+		case agent.hp >= t.healthInfo.HealthyLevel:
+			switch {
+			case agent.hp >= t.healthInfo.HealthyLevel+int(0.95*float64(t.healthInfo.WidthHealthy)):
+				newHP = t.healthInfo.StrongLevel
+				agent.daysInState = 0
+			case agent.hp >= t.healthInfo.HealthyLevel+int(0.63*float64(t.healthInfo.WidthHealthy)):
+				newHP = int(math.Max(float64(t.healthInfo.HealthyLevel), float64(agent.hp)-0.63*float64(t.healthInfo.WidthHealthy)))
```

Comment:
Will the new HP not always be the right side value if you're in this case? Please correct me if I'm misunderstanding the code.

Reply:
I think this is true and could be changed to
```suggestion
				newHP = int(float64(agent.hp)-0.63*float64(t.healthInfo.WidthHealthy))
```

Reply:
@Rhys-J @SaraFFdez This was thought as a safeguard if for some reasons we change the parameters defining the levels' limits: for example, if we choose the weak level to be very narrow, substracting `0.63*float64(t.healthInfo.WidthHealthy))` from an agent being in the healthy level could lead this agent to wake up directly in the critical level. This would be the case if `WidthWeak` < `0.63*float64(t.healthInfo.WidthHealthy))` (with an agent starting on the lower border of the healthy level for example).
However, if we leave the parameters defining the levels as they are, this should never be an issue and we can simplify the code as you suggested. 
Let me know what you think about this, and I'll change the code accordingly if needed

Reply:
Nevermind, the safeguard only apply in the last case where
```
default:
				newHP = int(math.Max(float64(t.healthInfo.WeakLevel), float64(agent.hp)-0.63*float64(t.healthInfo.WidthHealthy)))
```
You are completely right about the line you mentioned. This is probably also the case in the other levels. I will correct the code tomorrow.